source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -4054,6 +4057,89 @@ void Graph::drawGraph(ImVec2 mousePos)
                 }
             }
         }
+        
+        // Build a nodegraph from the selected nodes.
+        if (ImGui::IsKeyReleased(ImGuiKey_C) && 
+            io2.KeyShift && 
+            !io2.KeyCtrl && 
+            _currUiNode != nullptr &&
+            !readOnly()) 
+        {
+            // Cut nodes
+            _copiedNodes.clear();
+
+            // Since we can't have a node graph inside a node graph, we check if we are already
+            // inside one or if any of the selected nodes is a nodegraph.
+            bool isNodeGraph = !_currUiNode->getNode() || _isNodeGraph;
+            if (!isNodeGraph)
+            {
+                isNodeGraph |= _currUiNode->getNode()->getImplementation()->isA<mx::NodeGraph>();
+            
+                for (ed::NodeId selected : selectedNodes)
+                {
+                    int pos = findNode((int) selected.Get());
+                    if (pos >= 0)
+                    {
+                        UiNodePtr node = _graphNodes[pos];
+                        
+                        if (node->getNode() != nullptr)
+                        {
+                            isNodeGraph |= node->getNode()->getImplementation()->isA<mx::NodeGraph>();
+                        }
+                        
+                        _copiedNodes.insert(
+                                std::pair<UiNodePtr, UiNodePtr>(_graphNodes[pos], nullptr));                                      
+                    }
+                }
+            }
+
+            // Delete nodes we just copied
+            if (!_copiedNodes.empty() && !isNodeGraph)
+            {
+                for (std::map<UiNodePtr, UiNodePtr>::iterator iter = _copiedNodes.begin(); iter != _copiedNodes.end(); iter++)
+                {
+                    UiNodePtr node = iter->first;
+                    deleteNode(node);
```

Comment:
This seems to break the links between nodes so even though they are copied the links are not copied over? 

Tried moving this later after the paste but it crashes with some shared ptr de-refencing error.  Even with this as is, I've seen hang in refresh if you try and select the surfaceshader in the default marble graph and try this operation then try and go back to the parent.

@lfl-eholthouser pinging you as you know this code much better than I.  If you don't delete then a nodegraph is created properly with linked nodes and have not experienced any side-effects.

Reply:
@kwokcb this works great, thank you ! 
I am noticing something though.
If I create an `add_float` then create another one by copy and pasting the first one, then select the copy and hit shift + C to create the subgraph, the node does not get cut (but also appears inside the subgraph). Clicking on the node then crashes the editor. The copy also is collapsed and doesn't show its inputs. 

This seems to be an issue with how pasting and cutting works. If I put down an `add_float` then copy it. Then select both and use ctrl+x to cut, the copy does not get cut.